File: UtilitarioEnvioData/Entidades/Log.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace UtilitarioEnvioData.Entidades
{
    public static class Log
    {
        public static void EscribirLog(string texto)
        {
            try
            {
                string directorio = AppDomain.CurrentDomain.BaseDirectory + "Log";
                string archivo = directorio + "\\Log.txt";

                if (!Directory.Exists(directorio))
                    Directory.CreateDirectory(directorio);

                StreamWriter streamWriter = new StreamWriter(archivo, true);
                streamWriter.WriteLine(DateTime.Now.ToShortDateString() + " " + DateTime.Now.ToShortTimeString() + " " + texto);
                streamWriter.Close();

            }
            catch (Exception ex)
            {

                EscribirLog(texto);
            }
        }
    }
}

File: Administrador de Contenidos/App_Code/Log.vb
Imports Microsoft.VisualBasic
Imports System.IO

Public Class Log

    Public Shared Sub ErrorLog(ByVal mensaje As String)
        Try
            Dim directorio As String = AppDomain.CurrentDomain.BaseDirectory & "Log"
            If Not Directory.Exists(directorio) Then
                Directory.CreateDirectory(directorio)
            End If

            Dim archivo As String = directorio & "\" & +Date.Now.Year.ToString("0000") & Date.Now.Month.ToString("00") & Date.Now.Day.ToString("00") + ".log"
            Dim streamWriter As StreamWriter = New StreamWriter(archivo, True)
            streamWriter.WriteLine(Date.Now.ToShortTimeString() & " " & mensaje)
            streamWriter.Close()
        Catch ex As Exception

        End Try
    End Sub

End Class
